tests/istio.mk

```diff
@@ -43,11 +43,12 @@ e2e_docker: push
 
 endif
 
+# If set outside, it appears it is not possible to modify the variable.
```

Comment:
I have a PR https://github.com/istio/istio/pull/2960 where I'm trying to improve things on the prow side.  The log from its run:
https://storage.googleapis.com/istio-prow/pull/istio_istio/2960/e2e-smoke/2213/build-log.txt
seem to show that the test did set E2E_ARGS:

<pre>E2E_ARGS=--test_logs_path=/home/bootstrap/go/src/istio.io/istio/_artifacts make e2e_all</pre>

and this was reflected in the recipe, e.g.:

<pre>go test -v -timeout 20m ./tests/e2e/tests/simple -args --test_logs_path=/home/bootstrap/go/src/istio.io/istio/_artifacts  --istioctl /home/bootstrap/go/out/lx/amd64/release/istioctl --mixer_tag a382cc1c0474dcf2a2a794d6f3749c0c8abfd85c --pilot_tag a382cc1c0474dcf2a2a794d6f3749c0c8abfd85c --ca_tag a382cc1c0474dcf2a2a794d6f3749c0c8abfd8
</pre>

Perhaps the difference is in whether E2E_ARGS is passed to make (as is done in CircleCI) vs set in the environment before calling make (as I'm trying to do in prow).  I originally did try to pass E2E_ARGS to make but make (despite my careful use of quotes) seems to have problems coping with the spaces when E2E_ARGS had multiple parameters (e.g., if the 2nd or later parameter in E2E_ARGS had a single-char parameter like "-s" then make would complain that it didn't understand -s).  So, I switched to populating E2E_ARGS in the environment instead.

I'm not sure if any of this might affect your approach, either here or in how make is invoked with parameters.

Reply:
Yes, I was confused as well - quite unexpected, I keep saying we must use as little of make as possible. I think what I did fixes the problem for both cases.